File: src/jjoller/foxinabox/Dealer.java
package jjoller.foxinabox;

import java.util.*;

/**
 * Is responsible for dealing the cards.
 */
public interface Dealer {

    void reset();

    void dealTableCards(TexasHand hand);

    void dealHoleCards(TexasHand.Player player);

    default int smallBlind() {
        return 1;
    }

    default int bigBlind() {
        return 2;
    }

}

File: src/jjoller/foxinabox/Dealer.kt
package jjoller.foxinabox

/**
 * Is responsible for dealing the cards.
 */
interface Dealer {

    fun reset()

    fun dealTableCards(hand: TexasHand)

    fun dealHoleCards(player: TexasHand.Player)

    fun smallBlind(): Int {
        return 1
    }

    fun bigBlind(): Int {
        return 2
    }
    
}

